src/viewport.cpp

```diff
@@ -1328,9 +1327,6 @@ void ViewportSign::UpdatePosition(int center, int top, StringID str, StringID st
 	this->center = center;
 
 	/* zoomed out version */
-	if (str_small != STR_NULL) {
```

Comment:
This breaks ViewportAddTownNames.

Reply:
That's annoying, it does. Some part of this was definitely flagged up as a redundant assignment, and I could've sworn I checked it properly...